src/transformers/modeling_utils.py

```diff
@@ -2093,7 +2094,35 @@ def save_pretrained(
 
         # Save the model
         if state_dict is None:
-            state_dict = model_to_save.state_dict()
+            # if model parameters are offloaded, onload and send state dicts to CPU
+            if hasattr(model_to_save, "_hf_hook") and isinstance(model_to_save._hf_hook, AlignDevicesHook):
+                state_dict = {}
+                for name, module in model_to_save.named_modules():
+                    if name == "":
+                        continue
+                    if hasattr(module, "_hf_hook") and isinstance(module._hf_hook, AlignDevicesHook):
+                        # onload meta tensors to execution device
+                        try:
+                            module._hf_hook.pre_forward(module)
+                        except MemoryError:
+                            print("Model must fit in CPU memory to call save_pretrained!")
+                        module_state_dict = module.state_dict()
+                        # offload meta tensors from execution device
+                        module._hf_hook.post_forward(module, torch.tensor([]))
+                    else:
+                        module_state_dict = module.state_dict()
+
+                    for key in module_state_dict:
+                        # ignore placeholder parameters that are still on the meta device
+                        if str(module_state_dict[key].device) == "meta":
+                            continue
+                        params = module_state_dict[key]
+                        # .to(device) copies tensor, leading to OOM
+                        params = send_to_device(params, "cpu")
```

Comment:
FYI accelerate still uses `.to(device)`, so this isn't really doing much here using `send_to_device` since it's already a tensor.

Reply:
Hmm I see your point, I am not sure why but I saw a memory leak (meaning a GPU memory increase) with `.to(device)` and not `send_to_device` for this step.  Glad we don't need this step anymore thanks to your suggestion!